### ISSUE TITLE
About 'self.gamma'

### ISSUE THREAD

**Issue by zhangguanghui1** [junfu1115/DANet#19]
Hi! I do not understand why set 'self.gamma' to be zero, in this way, "out‘  equals  'x', right? Thanks a lot!

"class PAM_Module(nn.Module):
    """Position attention module"""

    # Ref from Sagan
    def __init__(self, in_dim):
        super(PAM_Module, self).__init__()
        # self.channel_in = in_dim

        self.query_conv = nn.Conv2d(in_channels=in_dim, out_channels=in_dim // 8, kernel_size=1)
        self.key_conv = nn.Conv2d(in_channels=in_dim, out_channels=in_dim // 8, kernel_size=1)
        self.value_conv = nn.Conv2d(in_channels=in_dim, out_channels=in_dim, kernel_size=1)
        self.gamma = nn.Parameter(torch.zeros(1))

        self.softmax = torch.nn.Softmax(dim=-1)

    def forward(self, x):
        """
            inputs :
                x : input feature maps( B X C X H X W)
            returns :
                out : attention value + input feature
                attention: B X (HxW) X (HxW)
        """
        m_batchsize, C, height, width = x.size()
        proj_query = self.query_conv(x).view(m_batchsize, -1, width * height).permute(0, 2, 1)  # (H*W) * C//8
        proj_key = self.key_conv(x).view(m_batchsize, -1, width * height)  # C//8 * (H*W)
        energy = torch.bmm(proj_query, proj_key)  # (H*W) * (H*W)
        attention = self.softmax(energy)
        proj_value = self.value_conv(x).view(m_batchsize, -1, width * height)  # C * (H*W)

        out = torch.bmm(proj_value, attention.permute(0, 2, 1))  #
        out = out.view(m_batchsize, C, height, width)

        out = self.gamma * out + x
        return out
"

**Comment by TensorDL** [comment]
The gamma is a learnable parameter.
